apiserver/params/model.go

```diff
@@ -158,7 +158,8 @@ type ModelInfo struct {
 
 // ModelInfoResult holds the result of a ModelInfo call.
 type ModelInfoResult struct {
-	Result *ModelInfo `json:"result,omitempty"`
+	CAASModel *CAASModelInfo `json:"caas-model,omitempty"`
+	IAASModel *ModelInfo `json:"iaas-model,omitempty"`
```

Comment:
In this case Type is effectively implied by it being in a CAASModel or a IAASModel - the other option would be to redefine ModelInfo so that it has Type and then an indirect to CAASModel/IAASModel. I think that would be better, but means renaming ModelInfo -> IAASModelInfo. Probably not worth the effort for a prototype...